Move coverage generation for client monorepo to a new workflow job

rossjrw opened the pull request:
I noticed that our sunburst on [CodeCov](https://app.codecov.io/gh/scpwiki/wikijump/) only contained files from `ftml/` and `web/` but was missing `client/`.

I think I've managed to track this down to the [`pnpm test`](https://github.com/scpwiki/wikijump/blob/f6f8b6e0ebeb5b4ec42b4373205da2a80f07c946/.github/workflows/client.yaml#L39-L40) step. Tests are currently failing, so that step errors out, which results in subsequent steps not being executed, therefore coverage is never uploaded.

I'm not sure if this PR will fix it or not, but I'm hoping that it might prompt CodeCov to update our report and that might give us a definite answer. If this doesn't work it would probably be a good idea to split coverage generation out into another command (that doesn't fail), and see if that works instead.

Monkatraz commented:
I think this will fix it but it means that a test failure won't..., uh, fail? I think the proper fix would be to setup coverage to run in another workflow again, and to use the `|| true` thing there instead.

rossjrw commented:
Yeah - looks like this does work but _at great cost_

I'll revamp this PR to split out coverage generation.

rossjrw committed:
Remove nested nabs scripts

Nested nabs script declarations produce scripts starting with 'npm run',
which doesn't match what is currently in the package.json (which start
with 'npm'). This commit replaces any nesting with manual script
concatenation to ensure that all commands start with 'pnpm'.

This commit also replaces any nabs scripts combined with '&&' to YAML
sequences (except for the test runner, which will be changed in the next
commit), which will be more readable as more tasks are added to them.

rossjrw committed:
Split out coverage generation to new script

rossjrw committed:
Add coverage generation to client action

rossjrw commented:
Alright, looks we're back in business - coverage uploads correctly (so we're down to 63% from 74%), but tests still fail.

`pnpm cover` generates coverage; `pnpm test` just runs tests.

![image](https://user-images.githubusercontent.com/29130152/118568860-9e995080-b770-11eb-8561-c605aa85574e.png)

Monkatraz commented on .github/workflows/client.yaml:
```diff
@@ -42,6 +42,37 @@ jobs:
+        with:
+          path: |
+            ~/.pnpm-store
+          key: ${{ runner.OS }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
```
Cache collision here - set this to use something like `-pnpm-coverage-` for the cache key.

rossjrw replied:
Aha, good catch - thanks

```suggestion
          key: ${{ runner.OS }}-pnpm-coverage-${{ hashFiles('**/pnpm-lock.yaml') }}
```

rossjrw replied:
That's a shame, honestly - I would've hoped that the cache used for validation could also be used for coverage. I'm sure GitHub's servers can handle it lmao

Monkatraz reviewed (approved):
One change needed but approving just to get things moving.

I fixed tests in https://github.com/scpwiki/wikijump/pull/294/files btw

rossjrw committed:
Namespace PNPM cache to coverage

rossjrw committed:
Add coverage to restore keys

rossjrw committed:
Remove generic restore keys
